src/contact.rs

```diff
@@ -742,6 +742,9 @@ impl Contact {
     /// This name is typically used in lists.
     /// To get the name editable in a formular, use `Contact::get_name`.
     pub fn get_display_name(&self) -> &str {
+        if !self.authname.is_empty() {
```

Comment:
I think it is wrong: you need to try `name` first, and if it is not available, try `authname`. `authname` is what comes in the `From` header, and `name` is what you set manually. Issue #1014 describes it correctly. This `if` should come after the `name` check. And the actual bug is probably somewhere else, like something setting `name` to the `From` header contents.

Reply:
According to #1014, we need to reverse this order of `if`s *and* add a "name change" event, probably as a separate PR.

Reply:
So name should not be filles with anything unless the ui/user fills it to overwrite the authname?

Reply:
Yes, I think without any manual user action `name` is empty. It can only be modified by user manually.